File: ELearningBackEnd/src/main/java/com/example/elearningbackend/course/CourseReq.java
package com.example.elearningbackend.course;

import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class CourseReq {

    @NotBlank(message = "name is required")
    private String name;

    @NotBlank(message = "description is required")
    private String description;

    @NotBlank(message = "category is required")
    private long categoryId;

    @NotBlank(message = "Skill Level is required")
    private String skillLevel;

    @NotBlank(message = "price is required")
    private double price;

    private long instructorId;

    private String categoryName;

}

File: ELearningFrontEnd/src/app/dtos/course/course.req.ts
export class CourseReq{

    name: string;
    description: string;
    categoryId: number;
    skillLevel: string;
    instructorId: number;
    price: number;

    categoryName: string;
    constructor(data: any) {
        this.name = data.name;
        this.description = data.description;
        this.categoryId = data.categoryId;
        this.skillLevel = data.skillLevel;
        this.instructorId = data.instructorId;
        this.price = data.price;

        this.categoryName = data.categoryName;
    }
}